### ISSUE TITLE
Future Roadmap

### ISSUE THREAD

**Issue by Nilpo** [PiDrive/CommunityFoundationEdition#9]
Maintaining this project is pretty time intensive for the size and scope of the projects. The individual project spaces must be downloaded and rebuilt with every update and the entire project has to be updated when NOOBS is updated. The size of the repository is also growing because of the OS images that are contained in it. Because of these reasons, I'm considering a change to the repository and instead maintaining a build script that can be run to generate a current version of the project any time you wish. Pre-built releases will still be provided from time-to-time.

This would also allow for more flexibility. Users could customize their own build. For instance, you might prefer to use PINN as a base instead of NOOBS. Or you may wish to add more OSes such as Alpine or Balena Cloud, or ...insert your favorite RPi image here...

I'd like to hear your thoughts and suggestions on this.

I'm also going to be developing the Wiki. Mainly I plan on including instructions for building the project manually and creating NOOBS-compatible images from your favorite distro.

**Comment by sgrayban** [comment]
You show me how to make the builds and I'll do those and give you links to the builds to upload. That should take a lot of stress off of you.

Having users build those might start a barrage of tickets that will drive you crazy.

**Comment by Nilpo** [comment]
> You show me how to make the builds and I'll do those and give you links to the builds to upload. That should take a lot of stress of of you.
> 
> Having users build those might start a barrage of tickets that will drive you crazy.

That's true. That's why a build script would be the way to go. I have one mostly written already for personal use.

The biggest challenge is that there is no zip file (or any archive) of the NOOBS-ready OS images. Every file must be downloaded individually from the Raspberry Pi downloads web site. There used to be a few FTP mirrors, but they are no longer maintained and can't be relied upon for the latest files. So scripting the download (and optionally build process) is the most efficient way of going about things at this point. That's why I've begun doing it on my own for this repo. I'm just wondering if sharing the build script is a better approach. It will greatly reduce the repo size (and cost--I pay for Git LFS storage) and would allow anyone to build the project at any time even if I can't keep up with releases.

I do watch the Raspberry Pi OS and NOOBS projects. My intention was to produce a current build with every update but it has sort of slipped through the cracks this year given everything that has been going on.

**Comment by Nilpo** [comment]
> You show me how to make the builds and I'll do those and give you links to the builds to upload. That should take a lot of stress of of you.

I'm happy to show you how to do it if you want to help maintain this project. Passing files around may get to be a problem given the size of the finished SD card images. They're already too large to host on Github. I'm currently serving them from Amazon AWS. Though I imagine I could probably give you upload access if you wanted to start contributing.

**Comment by sgrayban** [comment]
Google my name and you will see I am no stranger to build scripts or coding in any language. Currently I develop device drivers for home automation written in Apache Groovy. I can help with any pointers you have on this project and add my own in the process.

**Comment by G3O-CH** [comment]
The idea looks great.
So it could be simpler to include special images like https://github.com/jprochazka/adsb-receiver/releases/tag/v2.6.3 ?
The ADSB-apps kill SD-cards and that's the reason I let them run on the raspi-SSD.
And perhaps one time the images will be updated...

I am not a coder, only an experienced user, but I will help here testing from time to time and giving feedback.

**Comment by Murderhead** [comment]
How can I add some more OS'es like Raspberry Pi OS (64 bit)[Beta] or TwisterOS.
